arangod/Aql/QueryList.cpp

```diff
@@ -309,38 +314,40 @@ std::vector<QueryEntryCopy> QueryList::listCurrent() {
   // so we reduce the possibility of having to reserve more room
   // later
   result.reserve(16);
-
-  size_t const maxLength =
-      _maxQueryStringLength.load(std::memory_order_relaxed);
+  absl::InlinedVector<std::shared_ptr<Query>, 16> queries;
```

Comment:
What is the reason to populate this vector in this method?
As far as I can see, it is only populated when iterating over all queries, but the vector contents are never used.
Instead of pushing the shared_ptr into this vector, we could as well continue with the shared_ptr itself, i.e.
```
      auto queryPtr = it.second.lock();
      if (!queryPtr) {
        continue;
      }
      auto& query = *queryPtr;
...
```

Reply:
https://github.com/arangodb/arangodb/pull/16111#pullrequestreview-940874099

Also its draft about this line https://github.com/arangodb/arangodb/pull/16113/files#diff-233b9a3cd7f4156d982e0803152e356f9d8dac03ddef8044172ebbe1c32f3343R1274

Reply:
Yes, main idea is move call dtor from under lock to without lock.
And you right, I should add comment

Reply:
https://github.com/arangodb/arangodb/pull/16113/commits/881e1b4f8639827b46787bf4f182684189aed20a

Reply:
https://github.com/arangodb/arangodb/pull/16111/commits/29f2268742ce10cab2e260f18dfefebedda467f5